### PR TITLE
Deleted update functions, getSingleNodeRootHash internal functions, a…

### LINKED ISSUE #2377: Delete update code from Lib_MerkleTrie
**Description**

Since removing the OVM, we no longer use the `update()` functionality in `Lib_SecureMerkleTrie` (for reference the call [occurred here](https://github.com/ethereum-optimism/optimism/blob/86a13dcb0cfc2050317797d031917a2585ce2a49/packages/contracts/contracts/optimistic-ethereum/OVM/verification/OVM_StateTransitioner.sol#L439)). 

We shouldn't have unused code lying around, so we should remove `Lib_SecureMerkleTrie.update()`, and any code which was used only by it.

**ToDo**

1. Delete [the `update()` function](https://github.com/ethereum-optimism/optimism/blob/3dee2651d720f492eb4b5d7fcebf2cd29f485afc/packages/contracts/contracts/libraries/trie/Lib_SecureMerkleTrie.sol#L48) in the secure trie lib.

2. Delete [the `update()` function](https://github.com/ethereum-optimism/optimism/blob/3dee2651d720f492eb4b5d7fcebf2cd29f485afc/packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol#L92) in the regular trie lib. 

3. Delete any other `internal` helper functions which were only used in `update()` and so are no longer used. (Keep deleting code until the compiler complains about it).

4. Delete any tests that tested the deleted code.

### PR DESCRIPTION
**Pull request by Michael-Vander-Meiden**
**Description**
packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol:
- deleted update function
- deleted getSingleNodeRootHash internal function

packages/contracts/contracts/libraries/trie/Lib_SecureMerkleTrie.sol:
- deleted update function
- deleted getSingleNodeRootHash internal function

packages/contracts/test/contracts/libraries/trie/Lib_MerkleTrie.spec.ts
- deleted test for update function

packages/contracts/test/contracts/libraries/trie/Lib_SecureMerkleTrie.spec.ts
- deleted test for update function
- deleted test for getSingleNodeRootHash internal function

**Metadata**
- Fixes #2377

### CODE REVIEW

**Comment by tynes** [comment]
Can you add a changeset for the contracts package and squash into a single commit? Thanks!

**Comment by Michael-Vander-Meiden** [comment]
> 

Sure thing, will do that now

**Comment by Michael-Vander-Meiden** [comment]
@tynes done!

**Comment by tynes** [comment]
Thanks for updating @Michael-Vander-Meiden! I apologize and should have mentioned that this should be a patch changeset. Pre 1.0.0 we aren't following semver exactly, do you mind updating this? Then it will be good to go

**Comment by Michael-Vander-Meiden** [comment]
> Thanks for updating @Michael-Vander-Meiden! I apologize and should have mentioned that this should be a patch changeset. Pre 1.0.0 we aren't following semver exactly, do you mind updating this? Then it will be good to go

Sorry I just saw this. Third time's the charm! Switched it to patch. 😁

**Comment by tynes** [comment]
cc @maurelian

File: packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol [hunk 1]
```diff
@@ -78,35 +78,6 @@ library Lib_MerkleTrie {
...
-        // Special case when inserting the very first node.
-        if (_root == KECCAK256_RLP_NULL_BYTES) {
-            return getSingleNodeRootHash(_key, _value);
-        }
-
-        TrieNode[] memory proof = _parseProof(_proof);
-        (uint256 pathLength, bytes memory keyRemainder, ) = _walkNodePath(proof, _key, _root);
-        TrieNode[] memory newPath = _getNewPath(proof, pathLength, _key, keyRemainder, _value);
-
-        return _getUpdatedTrieRoot(newPath, _key);
```
Comment (maurelian) [inline packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol:107]: You could also delete this function's implementation (and any subsequent functions called only by it).

**Review by maurelian — APPROVED** [review]
This is safe to merge once the checks are running, but it would be preferable to also delete `_getUpdatedTrieRoot()`.
If we can get that in quickly, great. But I'll approve regardless.

**Comment by mslipper** [comment]
Hi Michael!

To get builds running please unfollow your fork of `optimism` and follow the repo under our organization. CircleCI uses which repo you follow to determine which repo to build. More details from CircleCI's documentation:

> Note: If a user submits a pull request to your repository from a fork, but no pipeline is triggered, then the user most likely is following a project fork on their personal account rather than the project itself of CircleCi, causing the jobs to trigger under the user’s personal account and not the organization account. To resolve this issue, have the user unfollow their fork of the project on CircleCI and instead follow the source project. This will trigger their jobs to run under the organization when they submit pull requests.

File: packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol (hunk 1 above)
Comment (Michael-Vander-Meiden) [reply packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol:107]: done

**Comment by maurelian** [comment]
Ugh, sorry for the trouble @Michael-Vander-Meiden, but you'll need to rebase this on `develop`, and run `yarn lint` to [fix that failed check](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/1359/workflows/96d3ef0e-4bff-48e2-a92f-b922a33a88c5/jobs/4555/parallel-runs/0/steps/0-103).

Thanks for working through this with us.

**Comment by Michael-Vander-Meiden** [comment]
No worries, working on it now.

**Comment by Michael-Vander-Meiden** [comment]
hey @maurelian , just want to make sure I get this right. When you say rebase on `develop`, do you mean pull `develop` into my `iss2377` branch and then lint, or something else?

**Comment by maurelian** [comment]
`git pull --rebase origin develop` should be the command you need. The important thing is that your changes are made on top of the latest commit on our `develop`.

**Comment by Michael-Vander-Meiden** [comment]
Hey @maurelian ! There ended up being *a lot* more unused functions than I originally thought. I deleted those as well some more tests and fixed the linting issues.

If things still look good to you then I will go ahead and squash these commits to prep for merge.

**Comment by maurelian** [comment]
Nothing like the smell of deleted code in the morning!These changes look great, squash away.

**Comment by Michael-Vander-Meiden** [comment]
@maurelian looks ready for merge!😁

**Review by mslipper — APPROVED** [review]

**Pull request merged**